sessionctx/variable/sysvar.go

```diff
@@ -769,13 +769,13 @@ var defaultSysVars = []*SysVar{
 		return nil
 	}},
 	/* The following variable is defined as session scope but is actually server scope. */
-	{Scope: ScopeSession, Name: TiDBGeneralLog, Value: BoolToOnOff(DefTiDBGeneralLog), Type: TypeBool, skipInit: true, SetSession: func(s *SessionVars, val string) error {
+	{Scope: ScopeInstance, Name: TiDBGeneralLog, Value: BoolToOnOff(DefTiDBGeneralLog), Type: TypeBool, skipInit: true, SetGlobal: func(s *SessionVars, val string) error {
```

Comment:
These changes from `ScopeSession` to `ScopeInstance` are all correct, but for your first PR may I suggest you just change one setting from `ScopeSession` to `ScopeInstance`? You can then test it correctly.

The reason I suggest this, is that you will likely get a lot of merge conflicts by modifying all the variables in your first PR. We tend to make lots of small changes to system variables, and it might get a bit annoying for you.